src/main/java/seedu/address/logic/commands/TagAddCommand.java

```diff
@@ -0,0 +1,211 @@
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds a tag to a contact.
+ */
+public class TagAddCommand extends Command {
+
+    public static final String COMMAND_WORD = "add";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a tag to the contact. "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_TAG + "TAG...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_TAG + "owesMoney";
+
+    public static final String MESSAGE_ADD_TAG_SUCCESS = "Tag added";
+    public static final String MESSAGE_NO_SUCH_TAG = "this tag does not exist";
+    public static final String MESSAGE_TAG_ALREADY_ADDED = "The contact already has the tag.";
+
+    private final Index index;
+    private final Tag tag;
+    private final EditPersonDescriptor editPersonDescriptor;
+
+    /**
+     * Creates an AddCommand to add the specified {@code Person}
+     */
+    public TagAddCommand(Index index, Tag tag, EditPersonDescriptor editPersonDescriptor) {
+        requireNonNull(index);
+        requireNonNull(tag);
+        requireNonNull(editPersonDescriptor);
+
+        this.index = index;
+        this.tag = tag;
+        this.editPersonDescriptor = editPersonDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+        List<Tag> tagList = model.getTagList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        if (!model.hasTag(tag)) {
+            throw new CommandException(MESSAGE_NO_SUCH_TAG);
+        } else if (personToEdit.getTags().contains(tag)) {
+            throw new CommandException(MESSAGE_TAG_ALREADY_ADDED);
+        } else {
+            Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor, tag);
+            model.setPerson(personToEdit, editedPerson);
+            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+            return new CommandResult(String.format(MESSAGE_ADD_TAG_SUCCESS, editedPerson));
+        }
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code editPersonDescriptor}.
+     */
+    private static Person createEditedPerson(Person personToEdit, EditPersonDescriptor editPersonDescriptor,
+                                             Tag tag) {
+        assert personToEdit != null;
```

Comment:
Can requireNonNull be used here instead for the same purpose?

Reply:
I was not sure which was better, but I looked it up and found out that people prefer using requireNonNull. I shall make the changes. Thanks.